tests/test_core/test_layout.py

```diff
@@ -1238,3 +1160,43 @@ def Child():
                 }
             ],
         }
+
+
+async def test_render_removed_context_consumer():
```

Comment:
@Archmonger if you're curious the code below reproduces that detached model error

Reply:
Is it the nonlocals that are causing the issue?

Reply:
No, those are just a trick to get a handle on hook callback from outside the components. The comments in the test body hopefully explain the cause. If they aren't clear I can make a follow up PR to improve the description.